src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -549,7 +551,22 @@ public static IDisposable BuildEnhancedLinePragma(this CodeWriter writer, Source
             return NullDisposable.Default;
         }
 
-        return new LinePragmaWriter(writer, span.Value, context, characterOffset, useEnhancedLinePragma: true);
+        var sourceSpan = RemapFilePathIfNecessary(span.Value, context);
+        return new LinePragmaWriter(writer, sourceSpan, context, characterOffset, useEnhancedLinePragma: true);
+    }
+
+    private static SourceSpan RemapFilePathIfNecessary(SourceSpan sourceSpan, CodeRenderingContext context)
+    {
+        if (context.Options.RemapLinePragmaPathsOnWindows && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            // ISSUE: https://github.com/dotnet/razor/issues/9108
+            // The razor tooling normalizes paths to be forward slash based, regardless of OS.
+            // If you try and use the line pragma in the design time docs to map back to the original file it will fail,
+            // as the path isn't actually valid on windows. As a workaround we apply a simple heuristic to switch the
+            // paths back when writing out the design time paths.
```

Comment:
Is there a reason we didn't just change tooling to not do this normalization?

Reply:
Tooling has a *lot* of assumptions baked into it about this (it was actually what I initially considered doing)

What I think I'd like to do long term is track both a normalized path, and the original path as separate items on the span, and we can use whichever we need rather than doing this mapping backwards and forwards.